ttnn/cpp/ttnn/operations/normalization/softmax_backward/softmax_backward_pybind.cpp

```diff
@@ -0,0 +1,94 @@
+
+            Args:
+                softmax_output_tensor (ttnn.Tensor): The output tensor from the forward softmax operation.
+                grad_tensor (ttnn.Tensor): The gradient tensor with respect to the softmax output.
+                dim (int, optional): The dimension along which softmax was computed in the forward pass. Defaults to -1 (last dimension).
+
+            Keyword Args:
+                memory_config (ttnn.MemoryConfig, optional): Memory configuration for the output tensor. If not provided, inherits from input tensor.
+                compute_kernel_config (DeviceComputeKernelConfig, optional): Compute kernel configuration for the operation.
+
+            Returns:
+                ttnn.Tensor: Gradient tensor with respect to the original softmax input, with the same shape as the input tensors.
+
+            Note:
+                Both input tensors must have the same shape and be the result of a forward softmax operation and its corresponding gradient.
+
+            Supported dtypes and layouts
+
+            .. list-table::
+               :header-rows: 1
+
+               * - Dtypes
+                 - Layouts
+               * - BFLOAT16, BFLOAT8_B
+                 - TILE
+
+            Example:
+                .. code-block:: python
+
+                    # Forward pass
+                    input_tensor = ttnn.rand((1, 1, 32, 64), dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=device)
+                    softmax_output = ttnn.softmax(input_tensor, dim=-1)
+
+                    # Assume we have gradients from subsequent operations
+                    grad_output = ttnn.rand((1, 1, 32, 64), dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=device)
+
+                    # Backward pass
+                    grad_input = ttnn.softmax_backward(softmax_output, grad_output, dim=-1)
+    )doc";
+
+    using OperationType = decltype(ttnn::softmax_backward);
+
+    ttnn::bind_registered_operation(
+        module,
+        ttnn::softmax_backward,
+        doc,
+        ttnn::pybind_overload_t{
+            [](const OperationType& self,
+               const ttnn::Tensor& softmax_output_tensor,
+               const ttnn::Tensor& grad_tensor,
+               int32_t dim) -> ttnn::Tensor {
+                // Convert negative dimension to positive (Python convention: -1 means last dimension)
+                const auto rank = softmax_output_tensor.logical_shape().rank();
+                uint32_t normalized_dim = dim < 0 ? static_cast<uint32_t>(dim + rank) : static_cast<uint32_t>(dim);
+                return self(
+                    softmax_output_tensor,
+                    grad_tensor,
+                    normalized_dim /*, memory_config, compute_kernel_config, numeric_stable*/);
+            },
+            py::arg("softmax_output_tensor").noconvert(),
+            py::arg("grad_tensor").noconvert(),
+            py::arg("dim") = -1});
```

Comment:
as there are no overloads, it can be simple
```suggestion
        ttnn::pybind_arguments_t{
            py::arg("softmax_output_tensor").noconvert(),
            py::arg("grad_tensor").noconvert(),
            py::arg("dim") = -1});
```

Reply:
Done